### PR TITLE
[feature] Allow collecting ARP table information in DeviceData.data #34

### LINKED ISSUE #34: [monitoring] Allow collecting ARP table information in DeviceData.data
Like https://github.com/openwisp/openwisp-monitoring/issues/33, but with ARP table information.

Incoming ARP table data:

```python
{
    'type': 'DeviceMonitoring',
    'arp_table': [
        {
            'ip_address': '192.168.66.197',  # ipv4 or ipv6
            'mac_address': 'f2:f1:3e:56:d2:66',  # mac address
            'interface': 'br-lan',  # string
            'type': '0x1',  # string
            'flags': '0x2',  # string
            'mask': '*',  # string
        }
    }
}
```

The schema in `openwisp_monitoring.device.schema` will have to be updated  accordingly.

Once the data is saved and validated correctly (a few tests will need to ensure bogus arp info will be rejected) we will have to ensure the arp table info is shown in the device status tab (type, flags and mask can be ignored for the moment):
https://github.com/openwisp/openwisp-monitoring/blob/master/openwisp_monitoring/device/templates/admin/config/device/change_form.html#L66-L124

If the arp table info is not available, do not mention this in the admin UI.

Will need https://github.com/openwisp/lua-monitoring/issues/3 to be implemented in order to send the ARP information from the device.

### PR DESCRIPTION
**Pull request by PabloCastellano**
image

Closes #34

### CODE REVIEW

File: openwisp_monitoring/device/templates/admin/config/device/change_form.html [hunk 1]
```diff
@@ -63,6 +63,31 @@ <h2>{% trans 'Load status' %}</h2>
               </div>
               {% endif %}
               {% endif %}
+              {% if device_data.arp_table %}
+              <h2>{% trans 'ARP table' %}</h2>
+              {% for arp_entry in device_data.arp_table %}
+              <div class="inline-group">
```
Comment (nemesisdesign) [inline openwisp_monitoring/device/templates/admin/config/device/change_form.html:69]: as mentioned via IM, could you try using a <table> here?

Comment (PabloCastellano) [reply openwisp_monitoring/device/templates/admin/config/device/change_form.html:69]: Done. I have updated the screenshot

File: openwisp_monitoring/device/tests/test_models.py [hunk 2]
```diff
@@ -142,6 +161,18 @@ def test_clean_data_fail(self):
         else:
             self.fail('ValidationError not raised')
 
+    def test_validate_ip_address(self):
```
Comment (nemesisdesign) [inline openwisp_monitoring/device/tests/test_models.py:164]: I'd use a name like `test_validate_arp_data`

Comment (PabloCastellano) [reply openwisp_monitoring/device/tests/test_models.py]: Done

File: openwisp_monitoring/device/schema.py [hunk 3]
```diff
@@ -1,7 +1,10 @@
 # NetJSON DeviceMonitoring schema,
 # https://github.com/netjson/netjson/blob/master/schema/device-monitoring.json
 schema = {
-    "$schema": "http://json-schema.org/draft-04/schema#",
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/netjson/netjson/master/schema/device-monitoring.json",
```
Comment (nemesisdesign) [inline openwisp_monitoring/device/schema.py:5]: one question: do you know if this triggers the downoad of this file? or is it there just for documentation purposes?

**Pull request merged**